### PR TITLE
Improve error messaging when processsing modJson

### LINKED ISSUE #1351: [Feature Request]: Improve Spanner Change Stream To BigQuery error messaging
### Related Template(s)

Spanner_Change_Streams_to_BigQuery

### What feature(s) are you requesting?

The output of this task is to provide actionable error reporting within [FailsafeModJsonToTableRowTransformer](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/FailsafeModJsonToTableRowTransformer.java)'s context.

When `FailsafeModJsonToTableRowFn` in [FailsafeModJsonToTableRowTransformer](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/FailsafeModJsonToTableRowTransformer.java) invokes private method `modJsonStringToTableRow` it will try/catch in a generic `Exception`, concatenating `Exception::getMessage` which is nullable. Some users report the following in their logs: `Caught exception when processing element , storing into dead letter queue. null`

A fix may consider more specific Exception catches and diagnose/remediate why getMessage is null. Additionally, the error message could reference where the dead letter records are stored, based on `deadLetterQueueDirectory` or its default value when not provided.

### PR DESCRIPTION
**Pull request by damondouglas**
This PR closes #1351 logging exception causes to help inform errors when using the template.

### CODE REVIEW

File: v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/FailsafeModJsonToTableRowTransformer.java [hunk 1]
```diff
@@ -205,17 +210,37 @@ public void processElement(ProcessContext context) {
         }
       }
 
-      private TableRow modJsonStringToTableRow(String modJsonString) throws Exception {
```
Comment (shunping) [inline v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/FailsafeModJsonToTableRowTransformer.java:208]: any particular reason to change this function from private to default (no access modifier)?

Comment (damondouglas) [reply v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/spannerchangestreamstobigquery/FailsafeModJsonToTableRowTransformer.java:208]: Good call, thank you. I was experimenting with testing this and seeing if I could instantiate this on its own but required a lot of coupled unnecessary infrastructure. I forgot to revert it back to private.

**Comment by shunping** [comment]
LGTM!

**Comment by pradeepnr** [comment]
Hi @damondouglas , can you please let us know why this PR is not merged yet?

**Comment by pradeepnr** [comment]
Hi @damondouglas , can you please update on this PR? Thanks.

**Comment by damondouglas** [comment]
@pradeepnr Thank you for checking in. I see the following when I view the PR which is why I can't merge.

3yChUw9SqBdBuPw

**Comment by pradeepnr** [comment]
Hi @damondouglas, I see it requires 1 approval. Can't @shunping help approving this PR? Thanks.

**Comment by pradeepnr** [comment]
Hi @damondouglas, is there a way for me to access Dataflow template built on your dev branch till this PR gets merged?

**Comment by shunping** [comment]
> Hi @damondouglas, I see it requires 1 approval. Can't @shunping help approving this PR? Thanks.

I just approved it.

**Review by shunping — APPROVED** [review]

**Comment by pradeepnr** [comment]
HI @damondouglas / @shunping , can you please let me know which build to use to try out?

**Pull request merged**